compiler/semexprs.nim

```diff
@@ -907,6 +907,10 @@ proc buildEchoStmt(c: PContext, n: PNode): PNode =
   result = semExpr(c, result)
 
 proc semExprNoType(c: PContext, n: PNode): PNode =
+  let isPush = hintExtendedContext in c.config.notes
+  if isPush: pushInfoContext(c.config, n.info)
+  defer:
```

Comment:
Don't use ``defer`` nor try finally. This code cannot throw anyway.

Reply:
done (ditto)